rpcs3/Emu/Cell/lv2/sys_usbd.cpp

```diff
@@ -525,24 +585,43 @@ usb_handler_thread::~usb_handler_thread()
 	handled_devices.clear();
 	open_pipes.clear();
 	usb_devices.clear();
+	usb_passthrough_devices.clear();
 
 	for (u32 index = 0; index < MAX_SYS_USBD_TRANSFERS; index++)
 	{
 		if (transfers[index].transfer)
 			libusb_free_transfer(transfers[index].transfer);
 	}
 
+#if LIBUSB_API_VERSION >= 0x01000102
+	libusb_hotplug_deregister_callback(ctx, callback_handle);
+#endif
+
 	if (ctx)
 		libusb_exit(ctx);
 }
 
 void usb_handler_thread::operator()()
 {
 	timeval lusb_tv{0, 0};
-
+	if (!hotplug_supported)
+	{
+		usb_hotplug_timeout = get_system_time() + 4'000'000ull;
+	}
 	while (ctx && thread_ctrl::state() != thread_state::aborting)
 	{
-		// Todo: Hotplug here?
+		const u64 now = get_system_time();
+		if (usb_hotplug_timeout && now > usb_hotplug_timeout)
+		{
+			// If we did the hotplug scan each cycle the game performance was significantly degraded, so we only perform this scan
+			// every 4 seconds.
+			// On systems where hotplug is native, we wait a little bit for devices to settle before we start the scan
+			perform_scan();
+			if (!hotplug_supported)
+			{
+				usb_hotplug_timeout = get_system_time() + 4'000'000ull;
```

Comment:
`usb_hotplug_timeout = hotplug_supported ? 0 : get_system_time() + 4'000'000ull;` and remove the usb_hotplug_timeout = 0 in perform_scan.

Reply:
Its intentional, the idea is it keeps scanning until it actually finds the hotplugged device

Reply:
ill check though, and see if that is actually necessary or not.

Reply:
Yep okay, seems to work just fine i have made that change now.